src/main/java/bio/terra/app/usermetrics/UserLoggingMetrics.java

```diff
@@ -11,10 +15,10 @@
  * API request will get grouped together even if they are set in different methods.
  */
 @Component
+@Scope(value = SCOPE_REQUEST, proxyMode = ScopedProxyMode.TARGET_CLASS)
 public class UserLoggingMetrics {
 
-  private static final ThreadLocal<HashMap<String, Object>> metrics =
-      ThreadLocal.withInitial(() -> new HashMap<>());
+  private final HashMap<String, Object> metrics = new HashMap<>();
```

Comment:
while you're here, this class should use `Map` instead of `HashMap` for the declared types, since the code should declare the interface type, not the implementation type.
```suggestion
  private final Map<String, Object> metrics = new HashMap<>();
```